crates/sncast/tests/e2e/deploy.rs

```diff
@@ -405,3 +410,251 @@ async fn test_json_output_format() {
+
+    let nonce = {
+        // Get nonce
+        let provider = create_test_provider();
+        let args = vec![
+            "--accounts-file",
+            "accounts.json",
+            "--json",
+            "account",
+            "list",
+        ];
+
+        let snapbox = runner(&args).current_dir(tempdir.path());
+        let output = snapbox.assert().success();
+
+        let value: Value = serde_json::from_str(output.as_stdout()).unwrap();
+        let account_address = value["my_account"]["address"].as_str().unwrap();
+
+        provider
+            .get_nonce(
+                BlockId::Tag(BlockTag::Latest),
+                Felt::from_hex(account_address).unwrap(),
+            )
+            .await
+            .unwrap()
+            .to_string()
+    };
+
+    let args = vec![
+        "--accounts-file",
+        "accounts.json",
+        "--account",
+        "my_account",
+        "deploy",
+        "--url",
+        URL,
+        "--constructor-calldata",
+        "0x1",
+        "0x1",
+        "0x0",
+        "--contract-name",
+        "Map",
+        "--nonce",
+        nonce.as_str(),
+    ];
+    let args = apply_test_resource_bounds_flags(args);
+
+    let snapbox = runner(&args)
+        .env("SNCAST_FORCE_SHOW_EXPLORER_LINKS", "1")
+        .current_dir(tempdir.path());
+    let output = snapbox.assert().success();
+
+    assert_stdout_contains(
+        output,
+        indoc! {
+            "
+            Success: Deployment completed
+
+            Contract Address:         0x0[..]
+            Class Hash:               0x0[..]
+            Declare Transaction Hash: 0x0[..]
+            Deploy Transaction Hash:  0x0[..]
+
+            To see deployment details, visit:
+            contract: [..]
+            class: [..]
+            declare transaction: [..]
+            deploy transaction: [..]
+            "
+        },
+    );
+}
+
+#[tokio::test]
+async fn test_deploy_with_declare_nonexistent_contract() {
+    let contract_path = duplicate_contract_directory_with_salt(
+        CONTRACTS_DIR.to_string() + "/map",
+        "put",
+        "with_redeclare",
+    );
+    let tempdir = create_and_deploy_oz_account().await;
+    join_tempdirs(&contract_path, &tempdir);
+
+    let args = vec![
+        "--accounts-file",
+        "accounts.json",
+        "--account",
+        "my_account",
+        "deploy",
+        "--url",
+        URL,
+        "--constructor-calldata",
+        "0x1",
+        "0x1",
+        "0x0",
+        "--contract-name",
+        "Map",
+        "--nonce",
+        "0x123456",
+    ];
+    let args = apply_test_resource_bounds_flags(args);
+
+    let snapbox = runner(&args)
+        .env("SNCAST_FORCE_SHOW_EXPLORER_LINKS", "1")
+        .current_dir(tempdir.path());
+    let output = snapbox.assert().success();
+
+    assert_stderr_contains(
+        output,
+        indoc! {
+            "
+            Command: deploy
+            Error: Invalid transaction nonce
```

Comment:
This error is should be improved. User should see something like 
```
Error occurred during contract declaration: ...
```
Currently it's unclear whether this error comes from declaration or deployment.